### ISSUE TITLE
Https services returns SSL received a record that exceeded the maximum permissible length

### ISSUE THREAD

**Issue by popopanda** [istio/istio#17395]
(NOTE: This is used to report product bugs:
  To report a security vulnerability, please visit <https://istio.io/about/security-vulnerabilities/>
  To ask questions about how to use Istio, please visit <https://discuss.istio.io>
)

**Bug description**
Was able to curl https resources previously for few months, however it stopped working without any changes.

Curling any external resources via HTTPS returns 

```
[root@centos-78d97f44b5-ss69v /]# curl -v https://google.com
* About to connect() to google.com port 443 (#0)
*   Trying 172.217.7.142...
* Connected to google.com (172.217.7.142) port 443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
* NSS error -12263 (SSL_ERROR_RX_RECORD_TOO_LONG)
* SSL received a record that exceeded the maximum permissible length.
* Closing connection 0
curl: (35) SSL received a record that exceeded the maximum permissible length.
```

curling 127.0.0.1:15000 returns:
```
curl localhost:15000/certs
{
 "certificates": []
}
```

pods without sidecars work fine. Also tried redeploying Istio

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
curling https://google.com should return 301/200

**Steps to reproduce the bug**
Deploy a pod with sidecar enabled
run `curl -v https://google.com`

**Version (include the output of `istioctl version --remote` and `kubectl version`)**
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-26T00:04:52Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.6-eks-5047ed", GitCommit:"5047edce664593832e9b889e447ac75ab104f527", GitTreeState:"clean", BuildDate:"2019-08-21T22:32:40Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```
```
client version: 1.2.2
citadel version: 1.2.2
egressgateway version: 1.2.2
egressgateway version: 1.2.2
galley version: 1.2.2
galley version: 1.2.2
ingressgateway version: 1.2.2
ingressgateway version: 1.2.2
pilot version: 1.2.2
pilot version: 1.2.2
pilot version: 1.2.2
pilot version: 1.2.2
pilot version: 1.2.2
policy version: 1.2.2
policy version: 1.2.2
sidecar-injector version: 1.2.2
sidecar-injector version: 1.2.2
sidecar-injector version: 1.2.2
telemetry version: 1.2.2
telemetry version: 1.2.2
```

**How was Istio installed?**
via helm charts

mTLS not enabled
cert-manager not enabled
includeIPRanges is set to only intercept VPC IP cidr block (192.168.0.0/24, and the cluster cidr block (10.100.0.0/16), excluding everything else

**Environment where bug was observed (cloud vendor, OS, etc)**
AWS - EKS

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.

**Comment by popopanda** [comment]
curling with ubuntu image returns:

```
root@ubuntu-dc4bdbcc8-xv5fz:/# curl https://google.com -v
* Rebuilt URL to: https://google.com/
*   Trying 172.217.15.110...
* TCP_NODELAY set
* Connected to google.com (172.217.15.110) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* error:1408F10B:SSL routines:ssl3_get_record:wrong version number
* stopped the pause stream!
* Closing connection 0
curl: (35) error:1408F10B:SSL routines:ssl3_get_record:wrong version number
```

**Comment by howardjohn** [comment]
Maybe related to https://github.com/istio/istio/issues/16458

**Comment by howardjohn** [comment]
Presumed fix in #16458. If thats not the case please let me know
